### PR TITLE
Only include valid apps in search with no_filter (bug 1128783)

### PR DESCRIPTION
**Pull request by robhudson**

### CODE REVIEW

File: mkt/search/tests/test_views.py [hunk 1]
```diff
@@ -796,6 +796,25 @@ def test_no_filter(self):
         res = self.client.get(self.url, data={'filtering': 0})
         eq_(len(res.json['objects']), 1)
 
+    def test_no_filter_excludes_deleted_or_disabled(self):
```
Comment (chuckharmston) [inline mkt/search/tests/test_views.py]: This should probably be two different tests. My rubric: if a failure in different parts would be  of a failure of a different code path, it should be housed in a different test.

**Comment by diox** [comment]
There are useful existing tests in `mkt/search/tests/test_filters.py`, maybe look if it's appropriate to add/modify a test there as well ?

**Comment by chuckharmston** [comment]
Great tests, I think this is the right fix for a kind of confusing bug. Nice job! r+

I think there are some other tests that might be affected in `mkt.search`, make sure the build passes before merging.

**Comment by robhudson** [comment]
Yep, I was waiting for Travis to tell me if I missed anything.

File: mkt/search/tests/test_views.py (hunk 1 above)
Comment (robhudson) [inline mkt/search/tests/test_views.py]: Ok. I'll separate the tests.

**Pull request merged**